PhysicsTools/PatAlgos/interface/ModifyObjectValueBase.h

```diff
@@ -0,0 +1,61 @@
+#ifndef __PhysicsTools_PatAlgos_ModifyObjectValueBase_h__
+#define __PhysicsTools_PatAlgos_ModifyObjectValueBase_h__
+
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include <string>
+
+class ModifyObjectValueBase {
+ public:
+ ModifyObjectValueBase(const edm::ParameterSet& conf) : 
```

Comment:
I'm thinking if this machinery can be instead moved to CommonTools/CandAlgos or UtilAlgos.
So that if needed, it could extend to regular (non-PAT) as well.
specialized ``` modifyObject(pat::Electron&)  ``` may have to go though (or the AOD class types added as well)